### PR TITLE
refactor(googlemap and geolocationmap): Use native google maps api to set zoom and center

### PR DESCRIPTION
**Pull request by erlbe**
OMEGA-282

## What is the reason for this change?

Change the way zoom and centering in the map works. This gives us more control over the interactions in the map.

### What is the expected result?

Map to work as normal :)

## How is it solved?

Use the native Google maps api to set zoom and center (via fitBounds) instead of google-map-react, which left the developers confused.

## Any changes/updates that may affect others?

|                       | Changed |
| --------------------- | ------- |
| Unit tests            |         |
| Dependencies          |         |
| Environment variables |         |
| Build scripts         |         |

## How can it be tested and verified?

- [x] Verify that GeoLocationMap works (/drift-og-vedlikehold/meld-fra-om-farlige-forhold/tre-over-linje)

### CODE REVIEW

File: packages/ui/src/components/GeoLocationMap/GeoLocationMap.tsx [hunk 1]
```diff
@@ -191,24 +191,11 @@ export const GeoLocationMap: React.FC<GeoLocationMapProps> = ({
...
-        </div>
-      )}
-    </div>
-  )
-
   return (
     <div data-testid="GeoLocationMap" className={classes} style={style}>
       <GoogleMap
         center={center ? center : undefined}
+        autoFitBoundsToContent={false}
```
[the remark this answers is not in the document]
Comment (erlbe) [reply packages/ui/src/components/GeoLocationMap/GeoLocationMap.tsx]: Done

**Review by corneliusgdahling — APPROVED** [review]
LGTM lets :shipit: ... nvm. need more reviewers ⏳

File: packages/ui/src/components/GoogleMap/GoogleMap.tsx [hunk 2]
```diff
@@ -49,25 +62,33 @@ export const GoogleMap: React.FC<GoogleMapProps> = ({
...
 
   useEffect(() => {
-    if (polygons) {
+    if (!googleMap || !googleMaps || !autoFitBoundsToContent) return
+    if (markers.length || polygons.length) {
+      const markerCoordinates: Coordinate[] = markers.map(m => m.coordinate)
+      const polygonCoordinates: Coordinate[] = polygons.reduce(
+        (prev, next) => [...prev, ...next],
+        []
+      ) // flatten
```
Comment (t0ff3r) [inline packages/ui/src/components/GoogleMap/GoogleMap.tsx:84]:
```js
const polygonCoordinates = flatten(polygons) // Ramda ftw!
```

File: packages/ui/src/components/GoogleMap/GoogleMap.tsx [hunk 3]
```diff
@@ -49,25 +62,33 @@ export const GoogleMap: React.FC<GoogleMapProps> = ({
...
-    : {
-        center: defaultCenter,
-        zoom: defaultZoom,
-      }
 
   useEffect(() => {
-    if (polygons) {
+    if (!googleMap || !googleMaps || !autoFitBoundsToContent) return
+    if (markers.length || polygons.length) {
+      const markerCoordinates: Coordinate[] = markers.map(m => m.coordinate)
```
Comment (t0ff3r) [inline packages/ui/src/components/GoogleMap/GoogleMap.tsx:80]:
```js
const markerCoordinates: Coordinate[] = markers.map(prop('coordinate')) // Ramda FTW!
```

**Review by t0ff3r — APPROVED** [review]
Would prefer some more ramda love when it makes sense but other than that i looks good.

File: packages/ui/src/components/GoogleMap/GoogleMap.tsx (hunk 2 above)
Comment (erlbe) [reply packages/ui/src/components/GoogleMap/GoogleMap.tsx]: Agreed!

File: packages/ui/src/components/GoogleMap/GoogleMap.tsx (hunk 3 above)
Comment (erlbe) [reply packages/ui/src/components/GoogleMap/GoogleMap.tsx]: Agreed!

**Pull request merged**